client/lib/common/src/data/resource/resource-actions.js

```diff
@@ -1,29 +1,57 @@
 /** global window **/
 import {Actions} from 'flummox';
 let {localStorage} = window;
+import request from 'common/src/superagent';
+import {Services} from 'common/src/data/services';
 
 class ResourceActions extends Actions {
 
     saveResource(resource) {
-        localStorage.setItem(`resource.${resource.id}`, JSON.stringify(resource));
-        return resource;
+        return request
+                .put(`${Services.essentials.url}${Services.essentials.root}/${resource.id}`)
+                .type('json')
+                .accept('json')
+                .send(resource)
+                .exec()
+                .then()
+                .catch( err => {
+                    err.id = resource.id;
+                    throw err;
+                });
     }
 
     saveScope(resourceId, scope) {
-        localStorage.setItem(`resource.${resourceId}.${scope.id}`, JSON.stringify(scope));
-        return [resourceId, scope];
+        return request
+                .put(`${Services.essentials.url}${Services.essentials.root}/${resourceId}/scopes/${scope.id}`)
+                .type('json')
+                .accept('json')
+                .send(scope)
+                .exec()
+                .then( [resourceId, scope] )
+                .catch( err => {
+                    err.id = scope.id;
+                    throw err;
+                });
     }
 
     fetchResource(resourceId) {
-        return JSON.parse(localStorage.getItem(`resource.${resourceId}`));
+        return request
+                .get(`${Services.essentials.url}${Services.essentials.root}/${resourceId}`)
+                .accept('json')
+                .exec()
+                .then( res => res.body )
+                .catch( err => {
+                    err.id = resourceId;
+                    throw err;
+                });
     }
 
     fetchResources() {
-        return Object
-                .keys(localStorage)
-                .filter(key => key.startsWith('resource'))
-                .filter(key => key.split('.').length === 2)
-                .map(key => JSON.parse(localStorage.getItem(key)));
+        return request
+                .get(`${Services.essentials.url}${Services.essentials.root}`)
+                .accept('json')
+                .exec()
+                .then( res => res.body );
```

Comment:
Somewhere we should handle the error. For lists of resources we could do it in the router to keep to store more clean from FetchResults.